File: typescript/classTest.ts

class Person{
	private name:string
	private age:number

	constructor(name:string, age:number){
		this.name=name
		this.age=age
	}

	say(){
		console.log(this.name, this.age)
	}
}

function test() {
	var a = new Person("Stanley", 20)
	a.say()
}

test()

File: scala/classTest.scala

class Person(var name:String, var age:Int){

	def say():Unit = {
		println(this.name, this.age)
	}

	def add():Unit = {
		age += 1
		println(age)
	}
}

object Test{
	def main(args: Array[String]):Unit = {
		val a = new Person("Stanley", 20)
		a.say()
		a.add()
	}
}